packages/fuel-gauge/src/revert-error.test.ts

```diff
@@ -46,30 +49,48 @@ describe('Revert Error Testing', () => {
     ]);
   });
 
-  it('can validate_inputs [assertion failed]', async () => {
-    const INPUT_PRICE = bn(100);
+  it('can throw RequireRevertError [invalid price]', async () => {
+    const INPUT_PRICE = bn(0);
     const INPUT_TOKEN_ID = bn(100);
 
     await expect(async () => {
       await contractInstance.functions.validate_inputs(INPUT_TOKEN_ID, INPUT_PRICE).call();
-    }).rejects.toThrow(/RevertError: The script reverted with reason: AssertFailed/);
+    }).rejects.toThrow(RequireRevertError);
   });
 
-  it('can validate_inputs [invalid price]', async () => {
-    const INPUT_PRICE = bn(0);
-    const INPUT_TOKEN_ID = bn(100);
+  it('can throw RequireRevertError [invalid token id]', async () => {
+    const INPUT_PRICE = bn(10);
+    const INPUT_TOKEN_ID = bn(55);
 
     await expect(async () => {
       await contractInstance.functions.validate_inputs(INPUT_TOKEN_ID, INPUT_PRICE).call();
-    }).rejects.toThrowError(RequireRevertError);
+    }).rejects.toThrow(RequireRevertError);
   });
 
-  it('can validate_inputs [invalid token id]', async () => {
-    const INPUT_PRICE = bn(10);
-    const INPUT_TOKEN_ID = bn(55);
+  it('can throw AssertFailedRevertError', async () => {
+    const INPUT_PRICE = bn(100);
+    const INPUT_TOKEN_ID = bn(100);
 
     await expect(async () => {
       await contractInstance.functions.validate_inputs(INPUT_TOKEN_ID, INPUT_PRICE).call();
-    }).rejects.toThrow(RequireRevertError);
+    }).rejects.toThrow(AssertFailedRevertError);
+  });
+
+  it('can throw SendMessageRevertError', async () => {
+    await expect(async () => {
+      await contractInstance.functions.failed_message().call();
+    }).rejects.toThrow(SendMessageRevertError);
+  });
```

Comment:
I think this type of asynchronous code testing may result in a flaky test, just like it was spotted [here](https://github.com/FuelLabs/fuels-ts/issues/845). 

I've also experienced some tests with this syntax occasionally failing in my setup. 

According to [Jest docs](https://jestjs.io/docs/asynchronous#asyncawait), we can simply use:
```ts
await expect(foo()).rejects.toMatch('error');
```